custom_components/unifiprotect/binary_sensor.py

```diff
@@ -440,24 +344,52 @@ def _async_fire_motion_event(self) -> None:
             },
         )
 
-    @property
-    def extra_state_attributes(self) -> dict[str, Any]:
-        """Return the extra state attributes."""
-        return {**super().extra_state_attributes, **self._extra_state_attributes}
 
+class ProtectDiskBinarySensor(ProtectNVREntity, BinarySensorEntity):
+    """A UniFi Protect NVR Disk Binary Sensor."""
+
+    def __init__(
+        self,
+        data: ProtectData,
+        device: NVR,
+        description: ProtectBinaryEntityDescription,
+        index: int,
+    ) -> None:
+        """Initialize the Binary Sensor."""
+        description = copy(description)
+        description.key = f"{description.key}_{index}"
+        description.name = (description.name or "{index}").format(index=index)
+        self._index = index
+        self.entity_description: ProtectBinaryEntityDescription = description
+        super().__init__(data, device)
+
+    @callback
+    def _async_update_device_from_protect(self) -> None:
+        super()._async_update_device_from_protect()
+
+        if self.entity_description.key.startswith(_KEY_DISK_HEALTH):
```

Comment:
You can reverse the condition and return early, then outdent below now that this is broken up.